ear-production-suite-plugins/lib/src/communication/direct_speakers_metadata_sender.cpp

```diff
@@ -73,6 +73,7 @@ void DirectSpeakersMetadataSender::sendMetadata() {
         if (!ec) {
           std::lock_guard<std::mutex> lock(timeoutMutex_);
           lastSendTimestamp_ = std::chrono::system_clock::now();
+          data_.set_changed(false);
```

Comment:
It's been forever since I've looked at this stuff, am I right in thinking the lambda is executed after the message is successfully sent? Is there a race condition here whereby the data gets updated after the send is queued but before it's delivered, but the renderer never sees the change as the flag gets reset?

Reply:
I think you might be right. Regardless of when the lambda is called, we're doing an async send anyway so definitely the potential for this to go wrong.
Do you think the `data_.set_changed(false);` should be performed immediately after ` auto msg = getMessage();`? (i.e, in the synchrous bit of the code) and then do a `data_.set_changed(true);` on failure? I think this should be failsafe, and it wouldn't matter if `data_` was updated in the meantime